src/encoded/static/components/dataset.js

```diff
@@ -1879,6 +1879,15 @@ export const SeriesComponent = (props, reactContext) => {
         combinedTreatmentAmounts = treatmentAmounts.join(', ');
     }
 
+    // As of code-time, there is a chance disease term namess across biosamples could be different. So
+    // the code takes this into account
+    const diseases = [
+        ...new Set(context.related_datasets.reduce((accumulatedDiseases, currentRelatedDataset) => {
+            const diseasesTermNames = currentRelatedDataset.replicates.map((replicate) => replicate.library.biosample.disease_term_name);
```

Comment:
replicates don’t require a library, and libraries don’t require a biosample, so should you test for their existence here? I’m not sure for disease series objects if maybe these properties are basically guaranteed even if not required.